sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala

```diff
@@ -146,6 +146,22 @@ trait CharVarcharTestSuite extends QueryTest with SQLTestUtils {
     }
   }
 
+  test("char/varchar with null value for partitioned columns") {
```

Comment:
this test will fail temporary due to other bugs being fixed

Reply:
Got it. BTW, please add `SPARK-34233: ` prefix.

Reply:
let's wait for #31339